streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1176,31 +1176,32 @@ public void cleanUp() {
      * @throws InvalidStateStoreException if Kafka Streams is (re-)initializing or a store with {@code storeName} and
      * {@code queryableStoreType} doesn't exist
      */
+    @Deprecated
     public <T> T store(final String storeName, final QueryableStoreType<T> queryableStoreType) {
-        return store(storeName, queryableStoreType, false);
+        return store(StoreQueryParams.fromNameAndType(storeName, queryableStoreType));
     }
 
     /**
-     * Get a facade wrapping the local {@link StateStore} instances with the provided {@code storeName} if the Store's
+     * Get a facade wrapping the local {@link StateStore} instances with the provided {@link StoreQueryParams}.
+     * StoreQueryParams need required parameters to be set, which are {@code storeName} and if
      * type is accepted by the provided {@link QueryableStoreType#accepts(StateStore) queryableStoreType}.
+     * The optional parameters to the StoreQueryParams include {@code partition} and {@code staleStoresEnabled}.
      * The returned object can be used to query the {@link StateStore} instances.
```

Comment:
We should not explain how `StoreQueryParams` works -- this should to into `StoreQueryParams` JavaDocs -- we should just keep the first sentence of the paragraph.

Reply:
Done.